src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/ElasticSearchDAO.scala

```diff
@@ -1,8 +1,103 @@
 package org.broadinstitute.dsde.firecloud.dataaccess
 
+import org.elasticsearch.action.admin.indices.create.{CreateIndexRequest, CreateIndexRequestBuilder, CreateIndexResponse}
+import org.elasticsearch.action.admin.indices.delete.{DeleteIndexRequest, DeleteIndexRequestBuilder, DeleteIndexResponse}
+import org.elasticsearch.action.admin.indices.exists.indices.{IndicesExistsRequest, IndicesExistsRequestBuilder, IndicesExistsResponse}
+import org.elasticsearch.action.bulk.{BulkRequest, BulkRequestBuilder, BulkResponse}
+import org.elasticsearch.action.delete.{DeleteRequest, DeleteRequestBuilder, DeleteResponse}
+import org.elasticsearch.action.index.{IndexRequest, IndexRequestBuilder, IndexResponse}
+import org.elasticsearch.client.transport.TransportClient
+import spray.http.Uri.Authority
+import spray.json.JsObject
+
 /**
   * Created by davidan on 9/28/16.
   */
-class ElasticSearchDAO extends SearchDAO {
+class ElasticSearchDAO(servers:Seq[Authority], indexName: String) extends SearchDAO with ElasticSearchDAOSupport {
+
+  private val client: TransportClient = buildClient(servers)
+  private final val datatype = "dataset"
+
+  // if the index does not exist, create it.
+  override def initIndex = {
+    _recreateIndex(false)
```

Comment:
I'm not sure about "recreate" as a name because it leaves it alone if it exists, in this case.